api/src/main/java/net/kyori/adventure/key/Key.java

```diff
@@ -29,7 +29,23 @@
 import org.checkerframework.checker.nullness.qual.NonNull;
 
 /**
- * A key.
+ * An identifying object used to fetch and/or store unique objects.
+ *
+ * <p>A key consists of:</p>
+ * <ul>
+ *   <li> a namespace - in most cases this should be your plugin or organization name</li>
+ *   <li> a value - what this key leads to, e.g "translations" or "entity.firework_rocket.blast"</li>
+ * </ul>
+ *
+ * <p>Valid characters for namespaces and keys are [a-z0-9_.-].</p>
```

Comment:
This is incorrect -- the valid regexes for namespaces and values are different. You'll also want to enclose the regexes in a `{@code ...}` block, and perhaps link to a regex -- for example `<a href="https://regexr.com/5ibbm">{@code [a-z0-9_.-]+}</a>`